src/webots/nodes/WbRobot.cpp

```diff
@@ -190,11 +191,35 @@ void WbRobot::preFinalize() {
 
   if ((WbTokenizer::worldFileVersion() < WbVersion(2020, 1, 0) ||
        (proto() && proto()->fileVersion() < WbVersion(2020, 1, 0))) &&
-      mControllerArgs->value().size() == 1 && mControllerArgs->value()[0].contains(" ")) {
-    mControllerArgs->setValue(mControllerArgs->value()[0].split(" "));
-    parsingWarn(tr("Robot.controllerArgs data type changed from SFString to MFString in Webots R2020b. "
-                   "Splitting controllerArgs at space boundaries. "
-                   "You may need to update your PROTO and/or world file(s)."));
+      mControllerArgs->value().size() == 1 && mControllerArgs->value()[0].contains(' ')) {
+    // we need to split the controllerArgs[0] at space boundaries into a list of strings
+    // taking into account quotes and double quotes to avoid splitting in the middle of a quoted (or double quoted) string
+    QStringList arguments;
+    const QString args = mControllerArgs->value()[0].trimmed();
+    const int l = args.length();
+    bool doubleQuote = false;
+    bool singleQuote = false;
+    int previous = 0;
+    for (int i = 0; i < l; i++) {
+      if (!singleQuote && args[i] == '"' && ((i == 0) || args[i - 1] != '\\'))
+        doubleQuote = !doubleQuote;
+      else if (!doubleQuote && args[i] == '\'' && ((i == 0) || args[i - 1] != '\\'))
+        singleQuote = !singleQuote;
+      else if (args[i] == ' ' && !(singleQuote || doubleQuote)) {
+        arguments << args.mid(previous, i - previous).remove('"').remove('\'');
```

Comment:
```suggestion
        if (args[i - 1] != ' ')
          arguments << args.mid(previous, i - previous).remove('"').remove('\'');
```
Otherwise, if more than one space is used between arguments it will add an empty argument to the list.